Dockerfile

```diff
@@ -0,0 +1,7 @@
+FROM node:12.16.3
+
+WORKDIR /src
+ADD . /src
+
+RUN npm install
+RUN npm run compile
```

Comment:
Why you can't run `npm test` here?

Reply:
This now produces a container with build contracts which we can do anything with like `npm run test`. I can't do `RUN npm test` but I could use `ENTRYPOINT` but then we lose the before mentioned benefit 🤷🏻 
I assume we could later use this container to test scripts and things like that.

If you think it's a good idea to produce a container that only runs `npm run test` we can do it, but I dont see much benefit in that or I am misunderstanding something.

Reply:
hmm, very strange. is it same as we do for `payments-smart-contracts` There tests can be run and here it is even simpler because you don't need run `ganache-cli` in saparate process.

Reply:
Yes in theory we only need to run `npm run test` but we have no `node` runners as far as I can see and gitlab doesn't provide them for free. So unless I am missing something, this is easiest/cheapest way to do this

Reply:
> hmm, very strange. is it same as we do for `payments-smart-contracts` There tests can be run and here it is even simpler because you don't need run `ganache-cli` in separate process.

I think You are wrong. This was the case when we were using travis. We have no node runners(and don't think we even need any) on gitlab, therefore any non Go jobs need to be run via docker. We're doing the same in `payments-smart-contracts`, a base image is built when calling the `docker-compose`.

Reply:
I'm ok for that, my question is why we can't run `npm test` instead of `npm run compile` in that docker?

Reply:
because this is the image build time, i.e. this is the image build process. All actions such as tests and what not should not be done during build time, but inside the built container.